### ISSUE TITLE
Documentation request on best practices used.

### ISSUE THREAD

**Issue by W1nstar** [ngx-rocket/generator-ngx-rocket#343]
## I'm submitting a...

- [ ] Bug report  
- [ ] Feature request
- [x ] Documentation issue or request

## Current behavior
When I first read ngx-rocket description was using "community's best practices" I wondered wich best practices were those. I couldn't find any doc about that exact issue.

## Expected behavior
It'd be ultra nice if it could be linked on that same description, on the documentation below and exported in the .md docs that the generator puts in the .\docs\ directory.

## Environment

```
- generator version: X.Y.Z
- node version: X.Y.Z  <!-- run `node --version` -->
- npm version: X.Y.Z  <!-- run `npm --version` -->
- OS:  <!-- Mac, Linux, Windows -->
```

Others:

I've managed several projects without e2e, unit testing, code scaffolding or CI. I reached this project because I'm learning myself about those things, and I was looking for an angular seed who'd help me by offering me community's best practices on those ends. I feel those best practices should be linked by the docs on the project, because it's never bad to link knowledge. Plus, internally the starter generated project looks different from what angular tutorial taughts you.

**Comment by sinedied** [comment]
I agree that that it's never bad to link knowledge, and we're always trying to improve the docs 😉 

Actually it may not be stated explicitly as the sources of the best practices used in the project, but if you take a look at the coding guides included in the project you have almost all the info there, like for the example the Angular guide: https://github.com/ngx-rocket/starter-kit/blob/master/docs/coding-guides/angular.md. Same for typescript, html etc.

Most of the best practices are based first on the most legitimate sources, ie the team behind the techs used (Angular team, Typescript team, RxJS team...), for example we use the official Angular Style Guide as the first source of truth. Then we try to include best practices from renowned web actors like Google, Facebook and others (for example the PWA guide https://developers.google.com/web/progressive-web-apps/), then community best practices mostly based on github/medium shared knowledge. The latter may be improved as it's usually linked in the related issues to implement them, but the docs are not always updated (and that's a miss!) with the links.
Finally, we rely at last on our own experience with big (as in team size like 15+ devs, code size like 100k+ LOC and maintenance time like 5+ years) projects for some non-trivial inclusions (proxy support, Http client extensions, i18n...).

We would love to improve our docs and try to within the limited time frame we all have (see also https://github.com/ngx-rocket/generator-ngx-rocket/issues/318), and you're more than welcome if you're willing to help us on that matter ❤️ 

PS: As for the Angular tutorial, it's sometimes in contradiction with their own style guide as they make shortcuts to ease the learning curve 😉Once you've caught the basics, you can move on to more complicated app structures designed for big apps.

**Comment by W1nstar** [comment]
Ok, thanks you. I will read them thoroughly, since I can't make the simplest of things work on a freshly generated project. Got no problem making get requests through the backend proxy to my localhost api, but it's been impossible to make a simple post request.
